libbeat/docs/upgrading.asciidoc

```diff
@@ -1,160 +1,329 @@
+--
 
-See <<breaking-changes-import-dashboards>> in the Breaking Changes
-documentation.
+[[non-es-outputs]]
+==== How to use versioned templates when the output is not {es}
 
-NOTE: If you've modified the Beats dashboards manually, reimporting them will
-overwrite your changes. Consider saving your dashboards under new IDs, so
-that the reimport doesn't overwrite them.
+If you've configured {beats} to send events to a different output, such as {ls},
+make sure you use versioned templates and indices. Otherwise, after you
+upgrade, there will be field conflicts. To use versioned index templates and
+indices, use the metadata from {beats} to set the index, and allow {beats} to
+manage the template.
 
-==== Heartbeat package names
+For example, for {ls}, set the {ls} `index` setting as recommended in the
+{logstash-ref}/plugins-inputs-beats.html[{beats} input plugin] documentation and
+allow {beats} to manage the index template:
 
-The DEB and RPM packages for Heartbeat are now named `heartbeat-elastic` to
-avoid conflicts with a different `heartbeat` project.  The `.deb` and `.rpm`
-file names haven't changed, just the package name in the repositories.
+[source,json]
+----
+    manage_template => false
+    index => "%{[@metadata][beat]}-%{[@metadata][version]}-%{+YYYY.MM.dd}" 
+----
 
-[[upgrade-mapping-template]]
-=== Upgrade the Elasticsearch mapping template
+When you use this configuration, the index name is set to match the index
+pattern in the {beats} index template.
+
+IMPORTANT: The index name must match the index pattern in the {beats} index
+template. For example, if {ls} sends events to an index called
++metricbeat-7-{localdate}+, but the index template expects indices to match
++metricbeat-{stack-version}-*+, you may encounter mapping errors and be unable
+to index {beats} events.
 
-Starting with Beats 6.0, the mapping templates and the default index names are
-versioned. For example, Metricbeat 6.0.0 creates indices like this:
+==== Upgrade dashboards
+
+We recommend that you re-import the {kib} dashboards after upgrading {kib} and
+{beats}. This way, you can take advantage of the new {beats} dashboards created
+for the 7.0 release. 
+
+Re-importing the dashboards will overwrite any manual changes that you've
+made. To avoid overwriting your changes, save any modified objects, such as
+dashboards, visualizations, and searches, under a new ID before proceeding. For
+more information about importing and exporting your modified objects, see
+{kibana-ref}/managing-saved-objects.html[Managing Saved Objects].
+
+To re-import the dashboards, run the `setup` command with the `--dashboards`
+option specified. For example:
 
 [source,shell]
-------------------------------------------------------------------------------
-metricbeat-6.0.0-2017-08-31
-------------------------------------------------------------------------------
+----
+metricbeat setup --dashboards
```

Comment:
If the index pattern in Kibana should also be compatible with 6.x data, also the migration flag has to be enabled here. You have a note below about the compatiblity layer for ingesting data, but it also applies here for the index pattern.

Reply:
Yeah, I moved things around a bit and sort of broke the order. It should mention the setting regardless of order, though, because we can't expect users to walk through these requirements in a specific order (unless we make them numbered steps, which I don't think we need to do.

Reply:
I'll look at this more closely when I run through everything again for testing purposes.